scipy/ndimage/tests/test_interpolation.py

```diff
@@ -54,77 +69,89 @@ def shift(x):
          ('constant', [-1, 1, 2, 3]),
          ('grid-constant', [-1, 1, 2, 3])]
     )
-    def test_boundaries2(self, mode, expected_value):
+    def test_boundaries2(self, mode, expected_value, xp):
         def shift(x):
             return (x[0] - 0.9,)
 
-        data = np.array([1, 2, 3, 4])
-        assert_array_equal(
-            expected_value,
+        if is_cupy(xp):
+            pytest.xfail("CuPy does not have geometric_transform")
+
+        data = xp.asarray([1, 2, 3, 4])
+        xp_assert_equal(
```

Comment:
let's fix `actual` and `desired` being the wrong way around here too

Reply:
OK, changed. In general, I agree with https://github.com/scipy/scipy/issues/21044 which calls for `_assert_` functions to be commutative (modulo rtol, but taht's a well-understood edge case IMO).

Reply:
yeah, but commutativity aside, the parameters are still called `actual` and `desired` 🙃

Reply:
FWIW, I don't consider it a big issue with being idiomatic. `assert_close(a, b)` and `assert_close(b, a)` should be equivalent. I'll fix it up if it it annoys you, though.